lib/collection/src/shards/transfer/shard_transfer.rs

```diff
@@ -30,9 +33,12 @@ pub struct ShardTransfer {
     /// If this flag is true, this is a replication related transfer of shard from 1 peer to another
     /// Shard on original peer will not be deleted in this case
     pub sync: bool,
+    /// Method to transfer shard with. `None` to choose automatically.
+    #[serde(default)]
+    pub method: Option<ShardTransferMethod>,
```

Comment:
The idea is to have automatic selection by default. It chooses the right transfer method based on the collection configuration.

For example, a small shard with no index would use stream records, while a large shard with indices would use snapshot transfer.

This is something to be implemented later.

What do you think?

Reply:
Got it, thanks for the explanation. Disregard this then.

Reply:
Maybe it's a good thing to revisit anyway. Because if using some automatic method, we could also add an `Auto` variant to make this behavior more explicit.